server/mentions.go

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) 2023-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/mattermost/mattermost-plugin-msteams/server/msteams"
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/plugin"
+)
+
+type UserNotification struct {
+	Trigger    string
+	Post       *model.Post
+	Channel    *model.Channel
+	PostAuthor *model.User
+	User       *model.User
+	Group      *model.Group
+}
+
+type NotificationsParser struct {
+	PAPI             plugin.API
+	Notifications    []*UserNotification
+	msteamsAppClient msteams.Client
+}
+
+func NewNotificationsParser(api plugin.API, msteamsAppClient msteams.Client) *NotificationsParser {
+	return &NotificationsParser{
+		PAPI:             api,
+		msteamsAppClient: msteamsAppClient,
+	}
+}
+
+func (p *NotificationsParser) ProcessPost(post *model.Post) error {
+	mentions := p.extractMentionsFromPost(post)
+	channel, err := p.PAPI.GetChannel(post.ChannelId)
+	if err != nil {
+		return err
+	}
+
+	for _, mention := range mentions {
+		m := &UserNotification{
+			Trigger: mention,
+			Post:    post,
+			Channel: channel,
+		}
+
+		switch mention {
+		case "@here":
+			fallthrough
+		case "@channel":
+			fallthrough
+		case "@all":
+
+		default:
+			user := p.isUserMention(strings.TrimPrefix(mention, "@"))
+			group := p.isGroupMention(strings.TrimPrefix(mention, "@"))
```

Comment:
Since these calls hit the database, and a mention can be only one of user or group here, can we avoid the `isGroupMention` call when it is a user mention?